sdk/azidentity/default_azure_credential.go

```diff
@@ -51,33 +54,42 @@ func NewDefaultAzureCredential(options *DefaultAzureCredentialOptions) (*Default
 	if err == nil {
 		creds = append(creds, envCred)
 	} else {
-		errMsg += err.Error()
+		errorMessages = append(errorMessages, fmt.Sprintf("EnvironmentCredential: %s", err.Error()))
 	}
 
 	msiCred, err := NewManagedIdentityCredential(&ManagedIdentityCredentialOptions{ClientOptions: options.ClientOptions})
 	if err == nil {
 		creds = append(creds, msiCred)
 		msiCred.client.imdsTimeout = time.Second
 	} else {
-		errMsg += err.Error()
+		errorMessages = append(errorMessages, fmt.Sprintf("ManagedIdentityCredential: %s", err.Error()))
 	}
 
 	cliCred, err := NewAzureCLICredential(&AzureCLICredentialOptions{TenantID: options.TenantID})
 	if err == nil {
 		creds = append(creds, cliCred)
 	} else {
-		errMsg += err.Error()
+		errorMessages = append(errorMessages, fmt.Sprintf("AzureCLICredential: %s", err.Error()))
 	}
 
+	errorMargin := "\n\t"
+	errorMessage := strings.Join(errorMessages, "\n\t")
+
 	if len(creds) == 0 {
-		err := errors.New(errMsg)
+		err := errors.New(errorMargin + errorMessage)
```

Comment:
This makes `err.Error()` return e.g. `\n\tEnvironmentCredential: ...`

Reply:
True! This is cleaner: https://github.com/Azure/azure-sdk-for-go/pull/16844/commits/6711742bcacfe30ac2714a385670addf0be65305

Should I make a test to verify the error message?

Reply:
Since this PR is for explicitly changing the error message string and the error types (thus messages) are well-known, yes please.  I did just that when making changing to the error string for `ResponseError`, see [here](https://github.com/Azure/azure-sdk-for-go/blob/main/sdk/azcore/internal/shared/response_error_test.go).

Reply:
Thank you! That makes sense to me 🙂

Reply:
@jhendrixMSFT what do you think of this approach? https://github.com/Azure/azure-sdk-for-go/pull/16844/commits/36b0c1de07b65dd5072744aa134d2ecc3d53ee83